quantum/dip_switch.c

```diff
@@ -52,13 +52,13 @@ void dip_switch_read(bool forced) {
     for (uint8_t i = 0; i < NUMBER_OF_DIP_SWITCHES; i++) {
         dip_switch_state[i] = !readPin(dip_switch_pad[i]);
         dip_switch_mask |= dip_switch_state[i] << i;
-        if (last_dip_switch_state[i] ^ dip_switch_state[i] || forced) {
+        if (last_dip_switch_state[i] != dip_switch_state[i] || forced) {
             has_dip_state_changed = true;
             dip_switch_update_kb(i, dip_switch_state[i]);
         }
     }
     if (has_dip_state_changed) {
         dip_switch_update_mask_kb(dip_switch_mask);
     }
-    memcpy(last_dip_switch_state, dip_switch_state, sizeof(&dip_switch_state));
+    memcpy(last_dip_switch_state, dip_switch_state, sizeof(dip_switch_state));
```

Comment:
👍

If I'm not mistaken, this line can be moved into the `if (has_dip_state_changed)` block above to avoid unnecessary copies.

Reply:
good! 
I'll test it in action and then commit it, just in case.

Reply:
done !